lib/src/components/NxTabs/NxTabs.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { Children, cloneElement, isValidElement, useMemo } from 'react';
+import classnames from 'classnames';
+
+import { Props as NxTabProps } from '../NxTab/types';
+import { Props as NxTabPanelProps } from '../NxTabPanel/types';
+
+import { Props, propTypes } from './types';
+export { Props } from './types';
+
+interface TabContextType {
+  activeTab?: number | null;
+  rootId: string;
+  index: number;
+  onTabSelect: (index: number) => void;
+};
+
+export const TabContext = React.createContext<TabContextType>({
+  activeTab: null,
+  rootId: '',
+  index: -1,
+  onTabSelect: () => {}
+});
+
+let tabId = 0;
+
+const NxTabs = function NxTabsElement(props: Props) {
+  const { activeTab, onTabSelect, id, className, children, ...attrs } = props;
+
+  const [tabList, ...tabPanels] = Children.toArray(children);
+
+  if (!isValidElement(tabList)) {
+    console.error('NxTabs must have an NxTabList');
+    return null;
+  }
+  else if (tabList.props.children.length === 0) {
+    console.error('NxTabs must have at least one NxTab');
+    return null;
+  }
+
+  const rootId = id || 'nx-tabs-' + useMemo(() => tabId++, []);
```

Comment:
You are correct, future calls of `useMemo` by React on the same instance of this component won't run which is how this prevents the tabId from simply incrementing each time you switch tabs or otherwise cause a re-render of this tab component. _Other_ instances will get a new tabId and increment the variable again. I don't really care if the variable changes for the same instance on a re-render other than it would be good to keep this tabId value small.

Reply:
Ah thanks for clarification, I was thinking the method reference was applying to all tabs and would only ever increment once